core/shared/src/main/scala/zio/ZQueue.scala

```diff
@@ -484,104 +483,118 @@ object ZQueue {
 
     val capacity: Int = queue.capacity
 
-    def offer(a: A): UIO[Boolean] = offerAll(List(a))
+    def offer(a: A): UIO[Boolean] =
+      UIO.effectSuspendTotal {
```

Comment:
We could optimize `effectSuspendTotal` primitive to do peekahead and immediately evaluate any `EffectTotal` / `Succeed` embedded inside there (such as the `succeedNow` in the below happy path).

Reply:
Actually that probably wouldn't make a difference, scratch that.